### PR TITLE
firedancer: fix epoch transition and pack topo

### PR DESCRIPTION
**Pull request by lheeger-jump**
Fixes an issue where we push stakes too early to poh tile and where pack gets corrupted frags and fixes some status cache issues.

### CODE REVIEW

**Review by arjain4 — APPROVED** [review]

File: src/disco/poh/fd_poh_tile.c [hunk 1]
```diff
@@ -42,6 +42,7 @@ fd_poh_tile_publish_tick( fd_poh_tile_ctx_t * ctx,
   ulong tspub = (ulong)fd_frag_meta_ts_comp( fd_tickcount() );
   ulong sz = sizeof(fd_entry_batch_meta_t)+sizeof(fd_entry_batch_header_t);
   ulong sig = fd_disco_poh_sig( slot, POH_PKT_TYPE_MICROBLOCK, 0UL );
+  FD_LOG_WARNING(("PUBLISH TICK TO SHRED: %lu %lu", tick->hashcnt_delta, tick->txn_cnt ));
```
Comment (ptaffet-jump) [inline src/disco/poh/fd_poh_tile.c:45]: Get rid of these warnings before merging, right?

Comment (arjain4) [reply src/disco/poh/fd_poh_tile.c:45]: We are actively debugging using this branch right now, making it a draft

File: src/disco/bank/fd_txncache.c [hunk 2]
```diff
@@ -1,5 +1,8 @@
 #include "fd_txncache.h"
 
+#pragma GCC diagnostic ignored "-Wformat"
```
Comment (arjain4) [inline src/disco/bank/fd_txncache.c:3]: let's get rid of this

File: src/disco/bank/fd_txncache.c [hunk 3]
```diff
@@ -349,33 +352,9 @@ fd_txncache_delete( void * shtc ) {
 static void
 fd_txncache_remove_blockcache_idx( fd_txncache_t * tc,
                                    ulong idx ) {
-  tc->blockcache[ idx ].lowest_slot = ULONG_MAX;
+  tc->blockcache[ idx ].lowest_slot = ULONG_MAX - 1UL;
```
Comment (lidatong) [inline src/disco/bank/fd_txncache.c:353]: need FD_TXNCACHE_XXX constants for the `(ULONG_MAX - 1UL)`, `-2UL` business that describe what's going on and documenting that 

i won't block this PR on that but please add

File: src/disco/bank/fd_txncache.c (hunk 2 above)
Comment (lheeger-jump) [reply src/disco/bank/fd_txncache.c]: done

**Review by lidatong — APPROVED** [review]

**Review by arjain4 — APPROVED** [review]

File: src/disco/bank/fd_txncache.c (hunk 3 above)
Comment (lheeger-jump) [reply src/disco/bank/fd_txncache.c:353]: @arjain4 can you do this in your slotcache PR?

Comment (arjain4) [reply src/disco/bank/fd_txncache.c:353]: yep have those changes in mind

**Pull request merged**